platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/sources/CustomGeometrySource.java

```diff
@@ -143,19 +145,49 @@ private void cancelTile(int z, int x, int y) {
     }
   }
 
+  @AnyThread
+  private ExecutorService obtainExecutorService() {
+    if (executorService == null || executorService.isShutdown()) {
+      createExecutorService();
+    }
+    return executorService;
+  }
+
+  @AnyThread
+  private void createExecutorService() {
+    executorService = Executors.newFixedThreadPool(4, new ThreadFactory() {
+      final AtomicInteger threadCount = new AtomicInteger();
+      final int poolId = poolCount.incrementAndGet();
+
+      @Override
+      public Thread newThread(@NonNull Runnable r) {
+        return new Thread(r, "CustomGeom-" + poolId + "-" + threadCount.incrementAndGet());
+      }
+    });
+  }
+
+  /**
+   * Clear out the used ExecutorService and remove the sleeping threads.
```

Comment:
If developers are required to call the `onStop` method when using this object, it should be called out explicitly here, and in the constructor or other documentation for `CustomGeometrySource`.